### PR TITLE
PF4 Chrome

### PR DESCRIPTION
**Pull request by seanforyou23**
This PR introduces PF4 Masthead. It's far from complete but serves as a good indicator of current progress.

Screen Shot 2019-04-25 at 5 53 23 PM

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/647/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/736/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/767/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/769/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/777/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/805/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/817/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/820/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by gashcrumb** [comment]
Great progress!  I noticed a few things while poking around the app:

nav on the side doesn't go to the bottom of the page:
image

The toolbar styling looks a bit odd, seems like a styling clash really between pf3 and pf4.

Am a little concerned that the app is gonna look washed out on a projector or on a screencast where there's usually color compression.  I don't think that's really in scope for this PR though to address, along with the toolbar styling.  We may need to get creative in a couple sprints :-)

**Comment by gashcrumb** [comment]
@dongniwang you should probably have a scan of the app too from the PR storybook link when you get a chance.

**Comment by dongniwang** [comment]
A few comments: 
- Are we not going to include the "?" icon and the menu items in this PR? They're currently missing from this version. 
- The vertical left nav, what's the font size? They look pretty small... And does the blue underline have a fixed width? I think both questions are also for @mcoker 
- And I noticed the same thing, the nav doesn't go all the way to the bottom of the page
- And the all white background (assuming that's coming from the masthead) makes it harder to read/scan content as what we have is based on PF3 (which has a grey background). @lboehling @mcoker what do you think?
- The "0 Results" part underneath the filter/sort toolbar should go away unless the filter/sort is being used. This could probably go in another PR.

**Comment by lboehling** [comment]
I think there needs to be a grey BKG per @dongniwang's comment. The drop shadows on the cards are angled in a way that makes the tops of the cards blend into the white background. More contrast is needed.

**Comment by gashcrumb** [comment]
> Are we not going to include the "?" icon and the menu items in this PR? They're currently missing from this version.

That's tracked as a separate item, so not for this PR.

> The "0 Results" part underneath the filter/sort toolbar should go away unless the filter/sort is being used. This could probably go in another PR.

Can you raise an issue for this?  I think this is just how the patternfly-react toolbar works though, which is what's being used across all of these views that have a toolbar.

**Comment by dongniwang** [comment]
@gashcrumb Cool, created the issue https://github.com/syndesisio/syndesis-react/issues/181 for the "# Results" thing.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/833/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by seanforyou23** [comment]
Thx for all the feedback! I've updated this PR to include what makes sense for this PR and can follow up with adding the help menu and whatever else comes up soon after we get this in.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/835/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
There is an issue with the wizard pages, the bottom footer is always pushed outside the viewport, regardless of it having enough space to show everything at once.

Wizards should never scroll, only their content should.

wizard-scroll

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/850/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by dongniwang** [comment]
Thanks for adding the background color! 

But now the Customizations page looks really weird. See below screenshot.  

Screen Shot 2019-04-26 at 11 22 36 AM

**Comment by riccardo-forina** [comment]
@dongniwang The bulk of the problem for that page is that we are abusing the tabs to implement horizontal navigation which, AFAIK, isn't a pattern in PF. Those links I think should stay on the sidebar, as a second level menu if we want to keep the "Customizations" entry or, be promoted to first level entries in the sidebar.

**Comment by dongniwang** [comment]
@riccardo-forina - good point! I think that deserves a wider discussion @syndesisio/uxd - what do you all think? 

cc: @gaughan

**Comment by seanforyou23** [comment]
Right, good feedback! So there's no doubt we'll find more styling related issues with the newly introduced pf4 stylesheets present. Some of these issues will warrant design considerations and some will require adjusting implementations for existing features. My hope is that with the most problematic styling issues already out of the way, we could get these changes merged and tackle the other issues as follow up. Otherwise, this PR could take quite a while to get these changes in and maintaining the branch against changes to master will continue to add to the workload. 

I've created issues for the items noted in the comments. Can we move the discussion there? WDYT?

**Comment by dongniwang** [comment]
@seanforyou23 - yeah, agreed. Happy to continue the discussion over there. For this one, everything looks good to me in terms of just the masthead and vertical nav. Thanks!

**Comment by gashcrumb** [comment]
> Wizards should never scroll, only their content should.

Agreed!  I think that's a real problem and the thing is, having this kind of disruption app-wide really makes it difficult then to develop pages, as I can't be certain if I'm the one screwing up the layout or if that's coming from the patternfly 4 styling that we have yet to fix.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/872/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/881/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by seanforyou23** [comment]
Alright, it's not perfect but the Wizard footer issue is fixed and I think it's at a good point in time to snapshot so we can evaluate the UX thoroughly and asses the PF3 in PF4 situation from this app's point of view. Let me know if there's anything else that needs to be addressed in this PR, otherwise, I think it's good to go.

File: app/ui-react/packages/ui/src/Data/Virtualizations/VirtualizationList.tsx [hunk 1]
```diff
@@ -68,6 +68,7 @@ export class VirtualizationList extends React.Component<
     return (
       <>
         <SimplePageHeader
+          className="container-fluid pf-u-my-md"
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/src/Data/Virtualizations/VirtualizationList.tsx]: Would it make sense to add this to the SimplePageHeader component to minimize the amount of custom classes added to these kinds of elements?

File: app/ui-react/packages/ui/src/Integration/CiCd/CiCdManagePageUI.tsx [hunk 2]
```diff
@@ -108,6 +108,7 @@ export class CiCdManagePageUI extends React.Component<
     return (
       <>
         <SimplePageHeader
+          className="pf-u-my-md"
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/src/Integration/CiCd/CiCdManagePageUI.tsx]: Same here, maybe we could add a prop similar to how `Button` lets you specify a size prop, I think it'd be good to minimize the user of `className` on components if possible.

**Pull request merged**